task/sql-tasks.js

```diff
@@ -281,7 +476,30 @@ async function task_1_21(db) {
  * @return {array}
  */
 async function task_1_22(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+        SELECT DISTINCT
+            Customers.CompanyName,
+            Products.ProductName AS "ProductName",
+            OrderDetails.UnitPrice AS "PricePerItem"
+        FROM Customers
+        INNER JOIN Orders
+            ON Customers.CustomerID = Orders.CustomerID
+        INNER JOIN OrderDetails
+            ON Orders.OrderID = OrderDetails.OrderID
+        INNER JOIN Products
+            ON Products.ProductID = OrderDetails.ProductID
+        WHERE OrderDetails.UnitPrice = (
+            SELECT
+                MAX(OrderDetails.UnitPrice) 
+            FROM Customers c
+            INNER JOIN Orders
+                ON Orders.CustomerID = c.CustomerID
+            INNER JOIN OrderDetails
+                ON Orders.OrderID = OrderDetails.OrderID
+            WHERE c.CompanyName = Customers.CompanyName)

```

Comment:
company name doesn't uniquely identify a customer. Also this field doesn't have an index so the query is slow